### PR TITLE
Suggested guidelines for minimum requirements for PRs

### PR DESCRIPTION
**Pull request by dsyme**
Some guidelines on what's needed for different kinds of PRs.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -39,9 +39,34 @@ All code submissions should be submitted with regression test cases, and will be
 
 - We expect contributors to be actively involved in quality assurance.
 - We expect contributors to be actively involved in quality assurance.
+- We expect contributors to be actively involved in quality assurance.
 - Partial, incomplete, or poorly-tested contributions will not be accepted.
 - Contributions may be put on hold according to stability, testing, and design-coherence requirements.
 
+#### Mimimum Bar for Code Cleanp Pull Requests
```
Comment (brodyberg) [inline CONTRIBUTING.md]: typo-> Cleanup

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -39,9 +39,34 @@ All code submissions should be submitted with regression test cases, and will be
 
 - We expect contributors to be actively involved in quality assurance.
 - We expect contributors to be actively involved in quality assurance.
+- We expect contributors to be actively involved in quality assurance.
```
Comment (brodyberg) [inline CONTRIBUTING.md]: Why is this line here 3x?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -39,9 +39,34 @@ All code submissions should be submitted with regression test cases, and will be
...
+- There must be no chance of a behavioural change, performance degradation or regression under any reasonable reading of the code in the context of the codebase as a whole.  
+- Code cleanup should generally be made separate to other checkins where possible, or the parts that are cleanup should be labelled as such.
+- Code cleanup is much more likely to be accepted towards the start of a release cycle. 
+
+#### Mimimum Bar for Performance Improvement Pull Requests
+
+Performance improvement checkins have the following minimum requirements (in addition to the above)
+
+- Performance tests and figures must be given, either in the PR or in the notes associated with the PR.  PRs without performance figures will be closed with a polite request to please add them.
+- The PR must show a reliable, substantive performance improvement tha justifies the complexity introduced (if any).  For the compiler, performance improvements of ~1% are of interest.  For the core library, it will depend on the routine in question. For the Visual F# tools, reactivity of the user interface will be of more interest than raw CPU performance.
```
Comment (brodyberg) [inline CONTRIBUTING.md]: typo: tha

File: CONTRIBUTING.md (hunk 2 above)
Comment (latkin) [inline CONTRIBUTING.md]: Because it's that important! :)

**Comment by latkin** [comment]
Did this create a new branch in the repo?

**Comment by latkin** [comment]
Also, :+1:

**Comment by dsyme** [comment]
@latkin -  yes, this created a new temporary branch. This is because I have "write" access to the repository and I edited the file directly in GitHub.  For people without write access, GitHub would fork the repo and create a PR when the edit is saved.  For people with write access, the options are either to commit directly or create a new temporary branch in the repo itself.  

I'll delete the branch in a bit, but in general you'll see this each time people with write access edit directly in github (something I find very useful)

File: CONTRIBUTING.md (hunk 2 above)
Comment (enricosada) [inline CONTRIBUTING.md]: like this seem a typo, maybe :
- **We expect contributors to be actively involved in quality assurance.**
- :exclamation:  We expect contributors to be actively involved in quality assurance.  
- :star: We expect contributors to be actively involved in quality assurance. :star:
